### DISCUSSION TITLE
batch Crop and resize bitmap images

### DISCUSSION THREAD

**Discussion by MrAdam1983** [ImageMagick/ImageMagick#5306] [Help]
Update - It appears that the BIMP plugin for GIMP will do what I need but I am curious if there is a process for Imagemagick.

I have 230 .bmp images with transparent backgrounds that I need to crop and resize.  I am using a software suite called Factorytalk View studio to develop touch screen interfaces for industrial equipment, the software only supports .bmp but can convert from .png and .jpeg.  I created my images as .png with transparency out of Fusion360 which is a bit limited on it's "image capture" abilities.  after conversion to .bmp my images are each about 30kb which I assumed was plenty small; of course they are not and I now understand that even though Factorytalk can scale images to fit on objects, it is actually a very tendinous task for the software to do regularly. 

Right now my images are all around 500x360 and I need to crop and scale 230 bitmaps to 120x60 so I don't have to repeat 16 hours of import work. 

For the "cropping" it looks `-trim` is the better option but I do not see how to get it to work without removing the transparency. 

I ran `magick mogrify -bordercolor None -border 1 -trim` on a copy of the folder containing my .bmps but it did not trim anything and it appears to have removed my transparency. 

Right now all of my Source .png files are in a specific folder structure to help organize them, I have confirmed that it I overwrite a file in the project folder of Factorytalk that the changes are picked up. So maybe I need to run a process against the .png versions and then convert them to .bmp?  I think as long as the name does not change I should be good.

**Comment by snibgo** [comment]
What version of IM, on what platform? Please always state this.

I suggest you first develop the command you need to convert one image. When that does what you want, either convert that to "magick mogrify", or put it in a shell loop to process all the files.

I am unclear what you need to do. Do you want to crop the images, or resize them, or what? Please post a sample input and required output.
